jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java

```diff
@@ -774,6 +778,12 @@ static void registerAll() {
 				.addInterface(openj9.lang.management.OpenJ9DiagnosticsMXBean.class)
 				.validateAndRegister();
 
+			/*[IF JAVA_SPEC_VERSION >= 21]*/
+			create(HOTSPOT_DIAGNOSTICS_MXBEAN_NAME, com.sun.management.internal.ExtendedHotSpotDiagnostic.getInstance())
```

Comment:
I think this needs a module check, like you can find in PlatformMBeanProvider.java. ExtendedHotSpotDiagnostic can only be created if the jdk.management module is loaded.

Reply:
Not completely sure. Let's try removing the changes from `ManagementUtils`. If testing shows side-effects, then I will add it back in `ManagementUtils`. In `PlatformMBeanProvider`, I have added a `jdk.management` module check for `HotSpotDiagnosticMXBean`.